lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -32,384 +57,165 @@ def get(section: str, option: str, fallback=None) -> str:
-            UnpackTarballs, "unpack", lambda self, *args: tickmock("unpack", None)
-        )
-        monkeypatch.setattr(UnpackTarballs, "update_symlink", mock_update_symlink)
-        obj = UnpackTarballs(MockConfig, make_logger)
-        result = obj.unpack_tarballs(self.min_size, self.max_size)
-        assert mocks_called == [
-            "glob",
-            "stat",
-            "stat",
-            "cachemanager",
-            "resolve",
-            "unpack",
-            "mock_update_symlink",
-            "resolve",
-            "unpack",
-            "mock_update_symlink",
-        ]
-        assert result.total == 2
-        assert result.success == 0
-
-    def test_unpack_tarballs_success(self, monkeypatch, make_logger):
-        """Show that when unpacking goes smoothly we can count the
-        successfully unpacked tarballs."""
-        mocks_called = []
-
-        def tickmock(id: str, ret_val: Any) -> Any:
-            mocks_called.append(id)
-            return ret_val
-
-        monkeypatch.setattr(
-            Path, "glob", lambda path, args: tickmock("glob", self.tarlist)
-        )
-        monkeypatch.setattr(
-            Path,
-            "stat",
-            lambda self, *args, **kwargs: tickmock(
-                "stat", TestUnpackTarballs.mock_stat(self, *args, **kwargs)
-            ),
-        )
-        monkeypatch.setattr(
-            Path, "resolve", lambda path, strict: tickmock("resolve", self)
-        )
-        monkeypatch.setattr(
-            CacheManager,
-            "__init__",
-            lambda self, config, logger: tickmock("cachemanager", None),
-        )
-        monkeypatch.setattr(
-            UnpackTarballs, "unpack", lambda self, *args: tickmock("unpack", None)
-        )
-        monkeypatch.setattr(
-            UnpackTarballs,
-            "update_symlink",
-            lambda path, *args: tickmock("update_symlink", None),
-        )
-        obj = UnpackTarballs(MockConfig, make_logger)
-        result = obj.unpack_tarballs(self.min_size, self.max_size)
-        assert mocks_called == [
-            "glob",
-            "stat",
-            "stat",
-            "cachemanager",
-            "resolve",
-            "unpack",
-            "update_symlink",
-            "resolve",
-            "unpack",
-            "update_symlink",
-        ]
-        assert result.total == 2
-        assert result.success == 2
+
+        with monkeypatch.context() as m:
+            m.setattr("pbench.server.unpack_tarballs.Path", MockPath)
+            m.setattr("pbench.server.unpack_tarballs.Sync", MockSync)
+            m.setattr("pbench.server.unpack_tarballs.CacheManager", MockCacheManager)
+            m.setattr(Metadata, "getvalue", getvalue)
+            obj = UnpackTarballs(MockConfig, make_logger)
+            tar = MockPath(datasets[0].tarball)
+            target = Target(datasets[0].dataset, tar)
+            MockCacheManager._fail_on([datasets[0].dataset.resource_id])
```

Comment:
Are you sure you want to fail (only) on the first dataset?  Wouldn't it be more interesting to fail on the second (or to run multiple trials with failures on different ones)?